pkg/providers/amazon/autoscaling/manager.go

```diff
@@ -129,3 +131,31 @@ func (m *Manager) GetAutoscalingGroupByStackName(stackName string) (*Group, erro
 
 	return NewGroup(m, describeAutoScalingGroupsOutput.AutoScalingGroups[0]), nil
 }
+
+// GetLastAutoscalingActivity returns the last ASG activity recorded.
+func (m *Manager) LastAutoscalingActivity(asgName *string) (*autoscaling.Activity, error) {
+	ctx, cancelFunction := context.WithTimeout(context.Background(), 15*time.Second)
+	defer cancelFunction()
```

Comment:
**Optional**: it would be more proper & idiomatic to take the context from the caller through a parameter in a cascadic fashion and only pass it here, sorry this was a quick solution during the pair programming when I wanted to check whether we can query the last ASG activity for failure reasons.

Reply:
Nice catch, done!